tests/tasks/create_wireless_profile_restart_network.yml

```diff
@@ -0,0 +1,21 @@
+# SPDX-License-Identifier: BSD-3-Clause
+---
+- name: Create wireless connection with rescue
+  block:
+    - name: Import network role
+      import_role:
+        name: linux-system-roles.network
+      vars:
+        network_allow_restart: '{{ wifi_restart_network }}'
+        network_connections:
+          - name: "{{ interface }}"
+            type: wireless
+            wireless:
+              ssid: "My WPA2-PSK Network"
+              key_mgmt: "wpa-psk"
+              password: "p@55w0rD"
+
+  rescue:
+    - name: Show rescue result
+      debug:
+        var: __network_service_restart_requested
```

Comment:
If you want this error to be raised to the caller, you'll need to have a task

```yaml
    - name: Raise error
      fail:
        msg: "{{ ansible_failed_result }}"
```

otherwise, the `rescue` will "eat" the error and the play will proceed normally - not sure if this is the desired behavior

Reply:
I just want to show the error message when the following task fails rather than stopping running the playbook : 

```
- name: Ask user's consent to restart NetworkManager due to wireless or team
    interfaces
  fail:
    msg: NetworkManager needs to be restarted to be able to proceed
      because wireless and team interfaces are defined. This might
      disturb the connectivity of the managed system. Please set
      `network_allow_restart` to `true` if you are prepared for this.
      Notice that the necessary action is to install NetworkManager-wifi or
      NetworkManager-team plugin and to restart NetworkManager.
  register: __network_service_restart_requested
  when:
    - __network_wireless_connections_defined
       or __network_team_connections_defined
    - network_provider == "nm"
    - not network_allow_restart
    - dnf_package_update_info is changed or yum_package_update_info is changed
    ```

Reply:
ok - then what you have is fine